numpy/add_newdocs.py

```diff
@@ -5525,6 +5525,37 @@ def luf(lamdaexpr, *args, **kwargs):
 
     """)
 
+add_newdoc('numpy.core.multiarray', 'format_float_OSprintf_g',
+    """
+    format_float_OSprintf_g(val, precision)
+
+    Print a floating point scalar using the system's printf function,
+    equivalent to:
+
+        printf("%.*g", precision, val);
+
+    for half/float/double, or replacing 'g' by 'Lg' for longdouble. This
+    method is designed to help cross-validate the format_float_* methods.
```

Comment:
Should this maybe go in the multiarray_tests.c.src instead (and not be exposed at all?)

Reply:
I'd like to expose it in case it helps users debug any problems.

Reply:
I mispoke - it would still be exposed through `numpy.core.multiarray_tests`

Reply:
Fair enough, will move.